src/commands/moderation/Slowmode.ts

```diff
@@ -46,11 +47,20 @@ export default class SlowmodeCommand extends BaseCommand {
 			interaction.options.getChannel("channel", false) ??
 			interaction.channel;
 
-		const time = /^\d+$/.test(timeString)
+		const isNumber = /^\d+$/.test(timeString);
+		const time = isNumber
 			? Number.parseInt(timeString)
 			: parse(timeString, "second") ?? 0;
 
-		if (!channel || !channel.isTextBased()) {
+		if (!isNumber && time === 0) {
+			interaction.reply({
+				content: "Invalid slowmode duration. Please provide a valid numerical duration (e.g., '5s', '1m', '1h').",
+				ephemeral: true,
+			});
+			return;
+		}
+
+		if (!channel || channel.type !== ChannelType.GuildText) {
```

Comment:
Channels of type other than GuildText also have slow mode. Use `channel.isTextBased()`.